docs/en/observability/profiling-upgrade.asciidoc

```diff
@@ -54,10 +56,359 @@ Integrations Server. To do this:
+  # We rely only on the "next" indices
+  delete_index_by_alias profiling-stackframes
+  delete_index_by_alias profiling-stacktraces
+  delete_index_by_alias profiling-executables
+
+  echo "Creating ILM policy..."
+  curl -s -f -o /dev/null -X PUT -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_ilm/policy/profiling-60-days" -H 'Content-Type: application/json' -d'
+{
+  "policy": {
+    "phases": {
+      "hot": {
+        "actions": {
+          "rollover": {
+            "max_primary_shard_size": "50gb",
+            "max_age": "30d",
+            "min_docs": 1
+          },
+          "set_priority": {
+            "priority": 100
+          },
+          "readonly": {}
+        }
+      },
+      "warm": {
+        "min_age": "30d",
+        "actions": {
+          "set_priority": {
+            "priority": 50
+          },
+          "forcemerge": {
+            "max_num_segments": 1
+          }
+        }
+      },
+      "delete": {
+        "min_age": "60d",
+        "actions": {
+          "delete": {}
+        }
+      }
+    },
+    "_meta": {
+      "description": "default policy for Elastic Universal Profiling",
+      "managed": true,
+      "version": 1
+    }
+  }
+}'
+
+  echo "Renaming indices to new naming scheme..."
+  rename_kv_index "profiling-stackframes"
+  rename_kv_index "profiling-stacktraces"
+  rename_kv_index "profiling-executables"
+
+  # profiling-symbols-private may or may not be present
+  status_code=$(curl -s -o /dev/null -w "%{http_code}" -I -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/profiling-symbols-private")
+  if [ "$status_code" == "200" ]; then
+    rename_regular_index "profiling-symbols-private"
+  fi
+
+  echo "Renaming existing fields..."
+  curl -s -f -o /dev/null -X PUT -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/profiling-events-*/_mapping" -H 'Content-Type: application/json' -d'
+  {
+    "properties": {
+      "profiling.project.id": {
+        "type": "alias",
+        "path": "service.name"
+      }
+    }
+  }'
+
+  echo "Enabling Universal Profiling index template management and disabling wildcard deletes..."
+  curl -s -f -o /dev/null -X PUT -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cluster/settings" -H 'Content-Type: application/json' -d'
+  {
+    "persistent": {
+      "xpack.profiling.templates.enabled": true,
+      "action.destructive_requires_name": true
+    }
+  }'
+
+  echo "Reenabling ILM..."
+  curl -s -f -o /dev/null -X POST -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_ilm/start"
+
+  printf "Checking cluster health ... "
+  local status=$(curl -s -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cluster/health?wait_for_status=green&wait_for_no_initializing_shards=true&wait_for_no_relocating_shards=true" | jq -r ".status")
+  if [ "$status" == "red" ]; then
+    printf "[ERROR: status is red]\n"
+  else
+    printf "[OK: status is %s]\n" "$status"
+  fi

```

Comment:
I tend to deploy my testing clusters without ES replication to save us some money & this check prevents me from upgrading such deployments. Should we allow yellow? Not sure. Perhaps alternatively we could add an argument to allow skipping this?

Reply:
Actually we do allow a `yellow` status. The script output should have been

> [OK: status is yellow]

And then the script should have continued normally. Did you see a different behavior?

Reply:
Ah, I was mislead by the snippet. There are actually two checks:

1. Before the migration to ensure the cluster is in a healthy state. Indeed, there we check for cluster health `green`
2. After the migration to make sure nothing is off. We are more lenient here and allow `yellow` because it's possible that replicas are not fully up yet after the migration (they will be eventually)

I think we can be probably a bit more lenient and allow cluster health `yellow` in the first check. I've pushed the change in https://github.com/elastic/observability-docs/pull/2995/commits/14f3e39f71edb2c26b1e35c45485d35c9f58bfd5.

Reply:
Ah, sorry for the confusion with quoting the wrong snippet.